### DISCUSSION TITLE
list available memcached commands

### DISCUSSION THREAD

**Discussion by ArtemIsmagilov** [memcached/memcached#1159] [General]
Please tell me how can I see all available commands in memcached? It seems that several commands are in development and are not available at all in other versions of memcached. The documentation describes all the commands for the latest version. Can we implement a command that would show a list of all available commands with their arguments, as in your documentation, but briefly?

```bash
memcached -C/--commands
```
return result
```text
get ...
gat ...
...
```

**Comment by dormando** [comment]
Maybe? Not sure how useful this is. Maybe a stats command that returns a bunch of one-liners?

Problems: 

- I don't think that's useful enough to justify my time on it.
- The list of commands changes very slowly, if ever. The base ascii commands haven't changed in over 10 years. The meta protocol has barely changed in 5 years. You very likely don't need to query memcached to see what it supports.

I'm not against the idea I just can't justify doing it.

**Reply by ArtemIsmagilov** [reply]
Thank you very much for answering)
